### DISCUSSION TITLE
Language model unavailable

### DISCUSSION THREAD

**Discussion by jarecsni** [community/community#142358] [Copilot Conversations] [answered]
### Select Topic Area

Question

### Body

Hi, this morning I received an email about the new O1 model being made available to me as part of the trial rollout on GitHub. However, when I tried to use it, I got the following error: 'language model unavailable.' Could someone pls advise on what steps I should take next to resolve this?

**Comment by wiredprairie** [comment]
Maybe this will help you (or someone else).

I had that error occur until I loaded a folder with code. I now see a small drop down to pick the model:

![image](https://github.com/user-attachments/assets/7ada2dcc-48dc-4443-b7ba-489330d4d944)

Now it just seems to work.

**Reply by jarecsni** [reply]
Thanks that’s useful. 

Any idea as to how to check if the Copilot plugin is really using o1? I got it working on another Mac and there when I ask copilot it says it’s running on GPT-4

**Reply by jarecsni** [reply]
Or any tell tale sign that o1 is being used would great thanks

**Reply by wiredprairie** [reply, marked as answer]
I just made sure it was selected in the drop-down as shown in the screen shot. When I asked chat, it responded either that it couldn't tell me or provided nothing useful. (Which admittedly was different from base 4o).

**Reply by jarecsni** [reply]
Thanks I didn’t spot that dropdown that’s a new new control in the Insiders version isn’t there in the stable release. And yeah it pretends I didn’t ask anything, it keeps saying hi I’m Copilot how can I help 😭

**Reply by kennedy-brrng** [reply]
If you can't see it yet try reload;

CMD + P, >Reload Window

**Reply by wedrfghndfshjibdcvgdf** [reply]
where can i find the copilot model switch button?
![image](https://github.com/user-attachments/assets/538aab16-1832-4a2a-921b-ad318d8adedc)

**Reply by wiredprairie** [reply]
Have you confirmed that you're running the correct version of Visual Studio Code or the Insiders build? Also, you need to have the preview version of the plugin. If those things are confirmed, you may want to log-out and log-in to your account to make certain everything is connected correctly. 

Otherwise, I'd suggest you create a new issue rather than asking in an answered question so that someone else may see it and respond.

**Reply by wedrfghndfshjibdcvgdf** [reply]
my vscode version: 1.95, GitHub Copilot Chat:
v0.23.2024102903 (pre-release), GitHub Copilot
v1.243.0

I've tried re-logging into the account and reinstalling vscode, reinstalling the plugin

**Reply by adityakakdeabb** [reply]
> where can i find the copilot model switch button? ![image](https://private-user-images.githubusercontent.com/183248821/381631414-538aab16-1832-4a2a-921b-ad318d8adedc.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzA4NzUzNDcsIm5iZiI6MTczMDg3NTA0NywicGF0aCI6Ii8xODMyNDg4MjEvMzgxNjMxNDE0LTUzOGFhYjE2LTE4MzItNGEyYS05MjFiLWFkMzE4ZDhhZGVkYy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTA2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTEwNlQwNjM3MjdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00NTFiZTI0NGE5ODE1ZjFiZDQzNjhjMGI5MGVhYWZkZTcwZTVlOTFmZjg5ODA1MjlmNzgyYzZkZGRlNzI5ZTBkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.o1v6S4vrrw4J7vFBcIzkutdbzfDgnokmritKUjYf2ik)

Been facing a similar issue ever since the update on Copilot. Unable to find the model selection dropdown in the chat box.

**Reply by RozenKristal** [reply]
For a paid product, it sure has a lot issue on updates.....

**Reply by RushilJalal** [reply]
I am facing the same issue but only when running code through WSL via VS Code Server.
The model selection dropdown is not visible and any entered prompts are ignored.

**Reply by Joyosmit** [reply]
Same issue as @RushilJalal . Copilot is providing auto suggestions, but Fix using Copilot is showing "Language model unavailable" on wsl.

**Comment by RozenKristal** [comment]
If your org is using a captive portal, this could potentially block the extension. I found out using sites like neverssl or similar to trigger it would help load the model for the chat feature. Just make sure you did it before opening copilot chat.

**Comment by kennedy-brrng** [comment]
Reload worked for me.

Kind regards,

Kennedy Nyaga | ​HelloDuty

helloduty.com

*+254 711 082-046*


*NB: This is not an automated email.​*

**Comment by RajaRameshChintaMobile** [comment]
Hi,

Currently i'm using MAC system.I'm facing **Language model unavailable issue** ,not able to fix the issue using above answer. I didn't see drop down also. My **VS code** version is : 1.96.2

**Reply by manh7890** [reply]
I have the same issue ,I didn't see drop down also. My VS code version is : 1.98.2

**Reply by learnwithalif** [reply]
i have same issue just now

**Reply by learnwithalif** [reply]
are the service down?

**Reply by learnwithalif** [reply]
<img width="430" height="258" alt="image" src="https://github.com/user-attachments/assets/25b8489f-82ab-4f97-a14a-cc39d5c4b6df" />

**Reply by KaisarFS** [reply]
same here with me, any solutions?

**Reply by tommyZZM** [reply]
same here too. but yesterday is fine. I tried reinstall but it doesn't work.

**Reply by andywang0607** [reply]
same here, my vs code version is 1.98.2 too
<img width="1083" height="134" alt="output" src="https://github.com/user-attachments/assets/fd27b95a-8da2-45af-95fe-4d3ec4e30591" />

<img width="832" height="121" alt="image" src="https://github.com/user-attachments/assets/97ac503e-e34b-4735-820a-bb9420cfb196" />

**Reply by AI-explorer-123** [reply]
same too

**Reply by zvivuz** [reply]
Just update your VSCode version and it will work.

**Reply by Mohamed-abul-fayasudeen** [reply]
Yes updating my VS code version has fixed this issue.

**Reply by jasonrichdarmawan** [reply]
The solution has issue. Some server is still use old version of Linux and so, not supported by VS Code after version 1.98.2

**Reply by YHao29** [reply]
Same issue, and I'm not allowed to update my vscode cz my server cannot support new version

**Reply by MichelPater** [reply]
Updating VS code worked for me as well. I had accidentally installed it in the Downloads directory on Mac making it unable to update. Moving it to the application folder allowed the update and fixed the Copilot integration.

**Comment by shabaan-qureshi** [comment]
I had this issue. Reinstalling the extensions did not work at all.

What finally did it for me was reinstalling VS Code again and changing the version to 1.102.2 from the one I was using (1.98)

**Comment by mohsenaahmadi** [comment]
like all other problem in computer, only need reload windows and be sure that you have access to github truly. (like dont sanction exist for you by IP)

**Comment by shalini-hm-sndk** [comment]
my copilot is taking longer time to give response
any idea?

**Comment by icancodefyi** [comment]
<img width="329" height="739" alt="image" src="https://github.com/user-attachments/assets/c361166a-c74c-44d1-b98d-297ff70e625d" />
any idea on this?

**Comment by toshanmugaraj** [comment]
Language model unavailable

**Reply by MarcusSpanierAdsuits** [reply]
same

**Reply by eric2788** [reply]
same

**Reply by icancodefyi** [reply]
why is this issue? any idea?

**Reply by Ownercz** [reply]
Can confirm, same with VS Code 1.107.1, but update is pending, so we shall see.

**Reply by eric2788** [reply]
> why is this issue? any idea?

here: https://www.githubstatus.com/

<img width="1032" height="554" alt="image" src="https://github.com/user-attachments/assets/8ca2a806-e190-4d0c-b33b-ccb58390c1e7" />

**Reply by Haugen** [reply]
Guessing Copilot is having some issues currently. I just ran into this as well. It worked 30 minutes ago.

**Reply by Dhruvil-Hexagon** [reply]
VS Code is already at the latest version, tried restarting IDE as well. Nothing worked

**Reply by YudheerRM** [reply]
Seems to be a global outage

**Reply by davixdedem** [reply]
> Guessing Copilot is having some issues currently. I just ran into this as well. It worked 30 minutes ago.

I think you got the point, global problem same here

**Reply by icancodefyi** [reply]
glad, i am not alone

**Reply by huanhuan6666** [reply]
I just encountered the same problem in VS Code 1.108.0.

**Reply by rakesh-vardan** [reply]
https://www.githubstatus.com/incidents/1lnqb2vk25vn

**Reply by rewyekha** [reply]
> VS Code is already at the latest version, tried restarting IDE as well. Nothing worked

GitHub Copilot backend is not accepting requests

VS Code sends requests → GitHub does not respond

VS Code shows:

Language model unavailable
PermissiveAuthRequiredError
Unexpected token '<'


GitHub returns an HTML error page instead of JSON
(exactly what your logs showed) 
<img width="1431" height="752" alt="image" src="https://github.com/user-attachments/assets/fc36135b-6410-41fd-9dc4-47fda069211e" />

**Reply by ravinderk** [reply]
I am facing same issue

**Comment by johnhyx** [comment]
<img width="578" height="338" alt="image" src="https://github.com/user-attachments/assets/683a2328-ce83-4926-aae4-0343b808e45c" />

**Reply by johnhyx** [reply]
版本: 1.107.1 (Universal)
提交: 994fd12f8d3a5aa16f17d42c041e5809167e845a
日期: 2025-12-17T14:15:14.850Z
Electron: 39.2.3
ElectronBuildId: 12895514
Chromium: 142.0.7444.175
Node.js: 22.21.1
V8: 14.2.231.21-electron.0
OS: Darwin arm64 24.6.0

**Reply by rakesh-vardan** [reply]
same for me with 1.108.0 also

**Comment by andang2208** [comment]
the same error, anybody know how to fix?

**Reply by toan10921** [reply]
same error too, i think github copilot have some bug and still updating

**Comment by MacdacMacdaxter** [comment]
The same thing for 15 minutes

**Comment by nitin2925** [comment]
same issue.

**Comment by akshay117** [comment]
same

**Comment by habensusz** [comment]
Same here, probably internal issue

**Comment by paimardpouria-oss** [comment]
Same here! ARGHHHH

**Comment by Nododo** [comment]
same 

why

**Reply by paimardpouria-oss** [reply]
Maybe internal issue. Or you have to un and install your VS Coder again

**Comment by gsponyai** [comment]
? me too, latest vscode

**Comment by atrakic** [comment]
https://www.githubstatus.com/

**Comment by devopsalaa** [comment]
Me to the same issue:  Language model unavailable

**Comment by samus-aran** [comment]
Hey @everyone from 2026 👋, we have announced an incident over on this [discussion](https://github.com/orgs/community/discussions/184156). The current discussion's situation was resolved and is unrelated to the current incident :) I will be locking this thread to reflect that.
